### ISSUE TITLE
Compact with threshold=-1.0 should be failed

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#3443]
**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
Do compact with invalid threshold=-1.0 response shouldn't be ok ranther than compact actually.
Test case is:
```
public void testCompactInvalidThreshold(MilvusClient client, String collectionName) {
        InsertParam insertParam = new InsertParam.Builder(collectionName).withFields(Constants.defaultEntities).build();
        InsertResponse res = client.insert(insertParam);
        assert(res.getResponse().ok());
        client.flush(collectionName);
        Response deleteRes = client.deleteEntityByID(collectionName, res.getEntityIds());
        assert (deleteRes.ok());
        client.flush(collectionName);
        Response resBefore = client.getCollectionStats(collectionName);
        System.out.println(resBefore.getMessage());
        CompactParam compactParam = new CompactParam.Builder(collectionName).withThreshold(-1.0).build();
        Response resCompact = client.compact(compactParam);
        assert (resCompact.ok());
        Response resAfter = client.getCollectionStats(collectionName);
        System.out.println(resAfter.getMessage());
    }
```
The collection stats before compact is:
```
{"data_size":7065580,"partitions":[{"data_size":7065580,"id":34,"row_count":0,"segments":[{"data_size":7065580,"files":[{"data_size":68120,"field":"_id","name":"_raw","path":"/C_"}...],"id":30,"row_count":0}],"tag":"_default"}],"row_count":0}
```
The collection stats after compact is:
```
{"data_size":0,"partitions":[{"data_size":0,"id":34,"row_count":0,"segments":null,"tag":"_default"}],"row_count":0}
```
**Steps/Code to reproduce behavior**
Follow this [guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**

**Method of installation**
- [ ] Docker/cpu
- [ ] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)

- Milvus version (master or released version)
master

**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**

**Additional context**

**Comment by shengjun1985** [comment]
The value should be in range [0, 1].

**Comment by yhmo** [comment]
Fixed by pr3495 pr3496 on master and 0.11.0 branch
